### ISSUE TITLE
Ng Hyderabad Meetup Ideas

### ISSUE THREAD

**Issue by elevenpassin** [coderplex/meetups#12]
Here are a couple of ideas that come to my mind when it comes to Angular and it's ecosystem. It's very general stuff, I may not even have any idea but I guess some topics 
are valuable.

Members from the Web Dev community are to suggest more ideas!

- Beginners session on typescript
- Getting started with Material design
- App walk through (basically explain how an app works and how to manage it etc)
- AngularJS design patterns (There’s observables, rxjs etc and maybe other patterns im not aware of)
- Serverside rendering and routing in angular
- Building a PWA with angular
- Testing angular apps
- File structure patterns, how to choose, pros and cons of different structures

**Comment by AhmedTa** [comment]
We already have many sessions on Angular and those were really useful but it would be great if we can have a session on following topics.

1) project architecture. How to plan the file structure based on type of project.
2) Angular web component (added in angular 6). How it can be used in real time scenario.
3) Session on observable pattern and Subject (different type of subject) in rxjs..

**Comment by dsasidhar** [comment]
Hey Guys,
 I am new to the community but have been working on angular for the past 1 year or so. I have a lot to learn and offer (I hope).

I think we should also try and have some sessions on NgRx. At our company we develop Angular with the goodness of Redux architecture. The NgRx community has grown well over the past one year. The things we can cover can be:

1. NgRx Store
2. NgRx Effects
3. Time machine debugging and Store dev tools.

**Comment by elevenpassin** [comment]
@dsasidhar Sounds great! We're happy to have you around. If you are interested in giving a talk about any of those topics, it would be amazing 💃

**Comment by dsasidhar** [comment]
@buoyantair Yes, I would be willing and interested to share the knowledge I have on these. I will be able to give an introduction, some light background and take it up to intermediate level I guess. This will be my first talk, so I will need help in understanding few things. Like, how much time should I be talking and do we expect live coding or simply presentation screens.

**Comment by elevenpassin** [comment]
@dsasidhar Awesome, The time depends largely on the topic you pick I guess and yes you can live code or present screenshots to convey the message :D

**Comment by dsasidhar** [comment]
@buoyantair super, let me know the timings and stuff.

**Comment by elevenpassin** [comment]
@dsasidhar Join out chatroom and we can discuss! https://chat.coderplex.org/
